docs/docs/migration/migration-from-v6-to-v7.md

```diff
@@ -136,38 +215,735 @@ It automatically installs the new versions of all `@comet` libraries, runs an ES
+    + },
+    ```
+
+2. Add CdnGuard
+
+    ```diff
+    // main.ts
+
+    + // if CDN is enabled, make sure all traffic is either coming from the CDN or internal sources
+    + if (config.cdn.originCheckSecret) {
+    +   app.useGlobalGuards(new CdnGuard({ headerName: "x-cdn-origin-check", headerValue: config.cdn.originCheckSecret }));
+    + }
+    ```
+
+3. Adjust `site/server.js`
+
 ```diff
-  UserPermissionsModule.forRootAsync({
-      useFactory: (...) => ({
-+         systemUsers: ["system-user"],
-          ...
-      }),
-      ...
-  }),
+// site/server.js
+
+- const cdnEnabled = process.env.CDN_ENABLED === "true";
+- const disableCdnOriginHeaderCheck = process.env.DISABLE_CDN_ORIGIN_HEADER_CHECK === "true";
+- const cdnOriginHeader = process.env.CDN_ORIGIN_HEADER;
++ const cdnOriginCheckSecret = process.env.CDN_ORIGIN_CHECK_SECRET;
+
+// ...
+
+- if (cdnEnabled && !disableCdnOriginHeaderCheck) {
+-    const incomingCdnOriginHeader = req.headers["x-cdn-origin-check"];
+-    if (cdnOriginHeader !== incomingCdnOriginHeader) {
++ if (cdnOriginCheckSecret) {
++    if (req.headers["x-cdn-origin-check"] !== cdnOriginCheckSecret) {
 ```
 
+4. DNS changes might be required. `api.example.com` should point to CDN, CDN should point to internal API domain
+
+### API Generator: Remove support for `visible` boolean, use `status` enum instead
+
+Replace the `visible` boolean field with a `status` enum field.
+Recommended enum values are (depending on the use case):
+
+-   Published/Unpublished
+-   Published/Unpublished/Archived
+-   Published/Unpublished/Deleted
+-   Active/Deleted
+-   Active/Archived
+
+The `update{Entity}Visibility` mutation is also removed.
+Use the generic `update{Entity}` mutation instead.
+
 ## Admin
 
+### Remove `axios` dependency
+
+Remove `axios` **if you don't use it in your project**.
+
 ### Rearrange components in `App.tsx`
 
 -   `ErrorDialogHandler` must be beneath `MuiThemeProvider` and `IntlProvider`
 -   `CurrentUserProvider` must be beneath or parallel to `ErrorDialogHandler`
+
+The resulting order should look something like this:
+
+```tsx
+// ...
+<IntlProvider locale="en" messages={getMessages()}>
+// ...
+    <MuiThemeProvider theme={theme}>
+    // ...
+        <ErrorDialogHandler />
+        <CurrentUserProvider>
+        // ...
+```
+
+### Rename `previewUrl` prop of `SiteConfig`
+
+The `previewUrl` prop of `SiteConfig` was renamed to `blockPreviewBaseUrl`.
+
+```diff
+- previewUrl = `${siteConfig.previewUrl}/page`;
++ previewUrl = `${siteConfig.blockPreviewBaseUrl}/page`;
+```
+
+### @comet/admin
+
+#### Remove the `@mui/styles` package
+
+The legacy `@mui/styles` package was removed in favor of `@mui/material/styles`.
+You can remove `@mui/styles` too:
+
+```diff
+//package.json
+
+- "@mui/styles": "^5.8.6",
+```
+
+This has multiple implications:
+
+-   COMET Admin components can now be styled using [MUI's `sx` prop](https://mui.com/system/getting-started/the-sx-prop/)
```

Comment:
I'm still not convinced that we should write Comet all caps in copy text 😬

Reply:
except when you want to yell at people :grinning:

Reply:
I don't feel strongly about this - a24e13b24b003d3a7e088a4bd1b253c668744480